### PR TITLE
feat(P-049): build multi-investment recovery and payoff insight surfaces

### PR DESCRIPTION
**Pull request by johnsherlock**
## Summary

- Extends `RangeInstallationContext` in `src/range/loader.ts` to expose `totalSystemInvestment` and `earliestAdditionDate` computed from `system_additions` records
- Adds `src/range/recovery.ts` with `computeAllTimeSavings` and `computePayoffOutlook` domain functions
- Updates `app/range/page.tsx` to load all-time savings (parallel query when history pre-dates the default 365-day window) and build a `RangeFinanceContext`
- Replaces the simple `PaybackTracker` in `RangeHistoryScreen` with an `InvestmentInsightsPanel` that surfaces three finance insights:
  - **Recovered so far** — cumulative savings vs total system investment, with partial-history caveat when record history predates synced data
  - **Repayment coverage this period** — period savings vs scheduled repayments (hidden for upfront-only portfolios)
  - **Approximate payoff outlook** — estimate shown only when ≥30 days of history and avg daily savings > 0; labeled explicitly as guidance
- Adds 18 focused unit tests covering happy paths, financing model variants (upfront-only, repayment-backed, mixed), partial-history omission, and boundary conditions

## Test plan

- [ ] All existing tests still pass (`vitest run`)
- [ ] TypeScript clean (`tsc --noEmit`)
- [ ] Range History shows `InvestmentInsightsPanel` when finance data is configured
- [ ] Prompt state shown when no finance records exist
- [ ] Repayment coverage hidden for upfront-only installations
- [ ] Payoff outlook hidden when < 30 days of history

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/web/app/range/page.tsx [hunk 1]
```diff
@@ -70,6 +80,32 @@ export default async function RangePage({ searchParams }: PageProps) {
       fixedCharges,
     );
 
+    let financeContext: RangeFinanceContext | null = null;
+    if (
+      installationContext?.totalSystemInvestment != null &&
+      installationContext.earliestAdditionDate != null &&
+      earliestDate != null
+    ) {
```
Comment (johnsherlock) [inline apps/web/app/range/page.tsx]: `P-049` says the setup prompt is for users **without investment records**, but this guard also requires `earliestDate != null`. A user who has configured system additions but has no persisted daily summaries yet will therefore get `financeContext = null` and see the "Set up investment" prompt anyway. That conflates "no investments" with "no recorded history yet", which the story treats as different states. A smaller corrective direction would be to build a finance context whenever investment records exist, then let the panel render a distinct no-history / waiting-for-data state when `coveredDays === 0`.

File: apps/web/src/range/loader.ts [hunk 2]
```diff
@@ -76,10 +81,23 @@ export async function loadRangeInstallationContext(
...
+    }
+
+    // Total system investment: upfront + (monthly × duration)
+    const upfront = a.upfrontPayment != null ? Number(a.upfrontPayment) : 0;
+    const monthly = a.monthlyRepayment != null ? Number(a.monthlyRepayment) : 0;
+    const duration = a.repaymentDurationMonths ?? 0;
+    totalInvestment += upfront + monthly * duration;
+
+    // Active monthly repayments (schedule not yet expired)
+    if (monthly <= 0 || !a.repaymentDurationMonths) continue;
```
Comment (johnsherlock) [inline apps/web/src/range/loader.ts]: This loader derives `activeMonthlyRepayment` relative to **now**, but `U-051`/`P-049` define repayment coverage as savings versus repayments due in the **selected period**. With the current shape, a historical range after a loan has ended will show no repayment coverage at all, and an older range during the repayment schedule will still use today's remaining-month state. The range screen needs period-aware repayment data here — for example by exposing raw repayment schedules or by computing repayments due for the requested range instead of a single current-month aggregate.

File: apps/web/app/range/RangeHistoryScreen.tsx [hunk 3]
```diff
@@ -675,16 +781,12 @@ function PaybackTracker({
     : 0;
 
   return (
```
Comment (johnsherlock) [inline apps/web/app/range/RangeHistoryScreen.tsx:790]: `P-049` now calls for repayment coverage to compare selected-period savings against repayments **due in that same selected period** for month, year, and custom range views. This proration uses `monthlyPayment * periodDays / AVG_DAYS_PER_MONTH`, which is only an approximation and will drift from the actual number of repayment months covered by the selected range, especially around partial months and calendar month boundaries. Once the loader exposes period-aware repayment schedule data, this should be replaced with an exact "repayments due in range" calculation rather than a day-based average.

**Review by johnsherlock — COMMENTED** [review]
Reviewed against `P-049` and `FE-014` at PR head `39c7cfed2be49b3c0ee95221fd613ed754a66203`. Beyond the existing unresolved Range History gating comment, three story-context gaps remain.
